commit 61fd3e0e7495f597b41e90d3e045b8c3b182a23d
Author: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date:   Tue May 31 12:28:20 2022 -0300

    socket: Use 64 bit stat for isfdtype (BZ# 29209)
    
    This is a missing spot initially from 52a5fe70a2c77935.
    
    Checked on i686-linux-gnu.
    
    (cherry picked from commit 87f1ec12e79a3895b33801fa816884f0d24ae7ef)

diff --git a/sysdeps/posix/isfdtype.c b/sysdeps/posix/isfdtype.c
index 06b5386c4379063d..f18bcfef224ebac6 100644
--- a/sysdeps/posix/isfdtype.c
+++ b/sysdeps/posix/isfdtype.c
@@ -24,12 +24,12 @@
 int
 isfdtype (int fildes, int fdtype)
 {
-  struct stat64 st;
+  struct __stat64_t64 st;
   int result;
 
   {
     int save_error = errno;
-    result = __fstat64 (fildes, &st);
+    result = __fstat64_time64 (fildes, &st);
     __set_errno (save_error);
   }
 
